lib/statsd_middleware.rb

```diff
@@ -31,16 +81,35 @@ def call(env)
     if path_parameters
       controller = path_parameters[:controller]
       action = path_parameters[:action]
+      source_app = get_source_app(env)
 
-      duration_tags = ["controller:#{controller}", "action:#{action}"]
-      status_tags = duration_tags + ["status:#{status}"]
-
-      # rubocop:disable Style/RescueModifier
-      StatsD.increment(STATUS_KEY, tags: status_tags) rescue nil
-      StatsD.measure(DURATION_KEY, duration, tags: duration_tags) rescue nil
+      instrument_statsd(status, duration, controller, action, source_app)
     end
 
-    # rubocop:enable Style/RescueModifier
     [status, headers, response]
   end
+
+  private
+
+  def get_source_app(env)
+    source_app = env['HTTP_SOURCE_APP_NAME']
+
+    return nil if source_app.nil?
+    return source_app if SOURCE_APP_NAMES.include?(source_app)
+
+    Raven.capture_message('Unrecognized Source App Request Header',
+                          level: 'warning', extra: { source_app_name: source_app })
+    nil
+  end
+
+  def instrument_statsd(status, duration, controller, action, source_app)
+    duration_tags = ["controller:#{controller}", "action:#{action}"]
+    duration_tags.push("source_app:#{source_app}") unless source_app.nil?
```

Comment:
We need to make sure that every metric sent to statsd with the same name has the same labels too. I'm not sure if `null` or an empty string is preferable to the library, but they should all have the same "shape" (i.e. labels).

Reply:
K. J/w - what happens if we don't?